shell/platform/darwin/common/framework/Source/FlutterChannelsTest.m

```diff
@@ -0,0 +1,143 @@
+  OCMExpect([binaryMessenger sendOnChannel:channelName
+                                   message:encodedMethodCall
+                               binaryReply:[OCMArg checkWithBlock:^BOOL(id obj) {
+                                 FlutterBinaryReply reply = obj;
+                                 reply(nil);
+                                 return YES;
+                               }]]);
+  [channel invokeMethod:@"foo"
+              arguments:@[ @(1) ]
+                 result:^(id _Nullable result) {
+                   didCallReply = YES;
+                   XCTAssertEqual(FlutterMethodNotImplemented, result);
+                 }];
+  OCMVerify(binaryMessenger);
+  XCTAssertTrue(didCallReply);
+}
+
+- (void)testMethodMessageHandler {
+  NSString* channelName = @"foo";
+  id binaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+
+  NSData* encodedMethodCall = [@"hey" dataUsingEncoding:NSUTF8StringEncoding];
+  OCMStub([codec encodeMethodCall:[OCMArg any]]).andReturn(encodedMethodCall);
+  FlutterMethodCallHandler handler =
+      ^(FlutterMethodCall* _Nonnull call, FlutterResult _Nonnull result) {
+        NSLog(@"hey");
+      };
+  [channel setMethodCallHandler:handler];
+  OCMVerify([binaryMessenger setMessageHandlerOnChannel:channelName
+                                   binaryMessageHandler:[OCMArg isNotNil]]);
+}
+
+- (void)testCallMethodHandler {
+  NSString* channelName = @"foo";
+  MockBinaryMessenger* binaryMessenger = [[MockBinaryMessenger alloc] init];
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+
+  NSData* encodedMethodCall = [@"encoded" dataUsingEncoding:NSUTF8StringEncoding];
+  NSData* replyData = [@"reply" dataUsingEncoding:NSUTF8StringEncoding];
+  NSData* replyEnvelopeData = [@"reply-envelope" dataUsingEncoding:NSUTF8StringEncoding];
+  FlutterMethodCall* methodCall = [[FlutterMethodCall alloc] init];
+  OCMStub([codec decodeMethodCall:encodedMethodCall]).andReturn(methodCall);
+  OCMStub([codec encodeSuccessEnvelope:replyData]).andReturn(replyEnvelopeData);
+  __block BOOL didCallHandler = NO;
+  __block BOOL didCallReply = NO;
+  FlutterMethodCallHandler handler =
+      ^(FlutterMethodCall* _Nonnull call, FlutterResult _Nonnull result) {
+        XCTAssertEqual(methodCall, call);
+        didCallHandler = YES;
```

Comment:
Use `-fulfill`s instead of didCallHandler-type locals.  Same for the above tests with this pattern.
```objective-c
XCTestExpectation* handlerCalled = [self expectationWithDescription:@"didCallHandler"];
FlutterMethodCallHandler handler =
      ^(FlutterMethodCall* call, FlutterResult result) {
        XCTAssertEqual(methodCall, call);
        result(replyData);
        [handlerCalled fulfill];
      };
[channel setMethodCallHandler:handler];
...
[self waitForExpectationsWithTimeout:1.0 handler:nil];
```

Reply:
done